superweather/superweather/LocationTool.h

```diff
@@ -0,0 +1,27 @@
+//
+//  LocationTool.h
+//  superweather
+//
+//  Created by mikael petersson on 2016-04-17.
+//  Copyright © 2016 mikael petersson. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <CoreLocation/CoreLocation.h>
+
+@protocol LocationListener <NSObject>
+
+- (void)newPosition: (CLLocation*)position;
+
+@end
+
+@interface LocationTool : NSObject<CLLocationManagerDelegate>
+
+@property (strong, nonatomic) CLLocationManager *locationManager;
+
+- (void) addListener: (NSObject<LocationListener>*) listener;
```

Comment:
Good idea to use the Observer pattern, but why not use NSNotifications? It's not so common in the objc community to implement it yourself. Check out this article on communication patterns in objc: https://www.objc.io/issues/7-foundation/communication-patterns/